x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FeatureImportance, TopClasses } from '../../../../../common/types/feature_importance';
+import { ExtendedFeatureImportance } from './decision_path_popover';
+
+export type DecisionPathPlotData = Array<[string, number, number]>;
+
+interface UseDecisionPathDataParams {
+  featureImportance: FeatureImportance[];
+  baseline?: number;
+  predictedValue?: string | number | undefined;
+  topClasses?: TopClasses;
+}
+
+interface RegressionDecisionPathProps {
+  baseline?: number;
+  predictedValue?: number | undefined;
+  featureImportance: FeatureImportance[];
+  topClasses?: TopClasses;
+}
+const FEATURE_NAME = 'feature_name';
+const FEATURE_IMPORTANCE = 'importance';
+
+export const useDecisionPathData = ({
+  baseline,
+  featureImportance,
+  predictedValue,
+}: UseDecisionPathDataParams): { decisionPathData: DecisionPathPlotData | undefined } => {
+  const [decisionPathData, setDecisionPlotData] = useState<DecisionPathPlotData | undefined>();
+
+  useEffect(() => {
+    const result = baseline
+      ? buildRegressionDecisionPathData({
+          baseline,
+          featureImportance,
+          predictedValue: predictedValue as number | undefined,
+        })
+      : buildClassificationDecisionPathData({
+          featureImportance,
+          currentClass: predictedValue as string | undefined,
+        });
+
+    setDecisionPlotData(result);
+  }, [baseline, featureImportance, predictedValue]);
```

Comment:
Since there is no async calls involved, maybe this can be done just using `useMemo` instead of both `useState` and `useEffect`.

Reply:
Updated in https://github.com/elastic/kibana/pull/73561/commits/ef815d11a2bf334d703366daff3519b38b848e11